risc0/zkvm/src/exec/mod.rs

```diff
@@ -277,7 +277,7 @@ impl<'a> Executor<'a> {
     ///
     /// This can be directly used by debuggers.
     pub fn step(&mut self) -> Result<Option<ExitCode>> {
-        if self.session_cycle() > self.env.get_session_limit() {
+        if self.session_cycle() > self.env.get_session_limit().unwrap_or(usize::MAX - 1) {
             return Ok(Some(ExitCode::SessionLimit));
         }
```

Comment:
```suggestion
        if let Some(limit) = self.env.get_session_limit() {
            if self.session_cycle() > limit {
                return Ok(Some(ExitCode::SessionLimit));
            }
        }
```

Reply:
While we're in here, how do you feel about me making the comparison `>=` instead of `>` ?

(That way, if someone sets limit to `usize::MAX`, we still get reasonable behavior.)

Reply:
sgtm